plugins/node/opentelemetry-instrumentation-net/src/net.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  diag,
+  Tracer,
+  Span,
+  SpanKind,
+  SpanStatusCode,
+} from '@opentelemetry/api';
+import {
+  InstrumentationBase,
+  InstrumentationConfig,
+  InstrumentationNodeModuleDefinition,
+  isWrapped,
+  safeExecuteInTheMiddle,
+} from '@opentelemetry/instrumentation';
+import { GeneralAttribute } from '@opentelemetry/semantic-conventions';
+import { Net } from './types';
+import { VERSION } from './version';
+import { platform } from 'os';
+import { Socket } from 'net';
+
+const IPC_TRANSPORT = platform() == 'win32' ? 'pipe' : 'Unix';
```

Comment:
```suggestion
const IPC_TRANSPORT = platform() === 'win32' ? 'pipe' : 'Unix';
```
please declare some const for 'pipe' and 'Unix' I would also keep names for them in upper case so `PIPE` and `UNIX`

Reply:
I took the attribute names directly from https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/span-general.md#general-network-connection-attributes where they have the same case, so I'm not sure which way to go here

Reply:
can you add then const values to general semantic conventions?.  I see there are already 3 values, you can add the one you need here
```
export const GeneralAttribute = {
  // These are used as potential values to NET_TRANSPORT
  ...
  UNIX: 'Unix',
  PIPE: 'pipe',
};

```

Reply:
here -> https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-semantic-conventions/src/trace/general.ts#L35

Reply:
Yeah, sure. Thanks :slightly_smiling_face:

Reply:
Although on another note, they are not really attributes, but attribute values. I guess it's fine to just declare the constants here?

Reply:
https://github.com/open-telemetry/opentelemetry-js/pull/2026

Reply:
I think for now you can create some const locally on top with `@TODO` note to replace them later with semantic conventions as they will be available after releasing

Reply:
Done: https://github.com/open-telemetry/opentelemetry-js-contrib/pull/389/commits/ac287749eb686eb9aac8455866f87deb7c7487f1

Also the other issues brought up should be resolved now as well.